spring-data-jdbc/src/test/java/org/springframework/data/jdbc/core/EntityRowMapperUnitTests.java

```diff
@@ -249,6 +249,54 @@ public void handlesNonSimplePropertyInConstructor() throws SQLException {
 		assertThat(extracted.content).hasSize(2);
 	}
 
+	@Test // DATAJDBC-359
+	public void chainedEntitiesWithoutId() throws SQLException {
+
+		ResultSet rs = mockResultSet(asList("four", //
```

Comment:
How about some mock/builder API? From just looking at the code, it's next to impossible to understand what's happening here. Check out [Cassandra's `RowMockUtil`](https://github.com/spring-projects/spring-data-cassandra/blob/42cc65aaab4d3d67546d815d3d9eb142a7d5c616/spring-data-cassandra/src/test/java/org/springframework/data/cassandra/core/convert/MappingCassandraConverterMappedTupleUnitTests.java#L70-L73).

Reply:
Done.